cfn-resources/test/e2e/search-deployment/searchdeployment_test.go

```diff
@@ -0,0 +1,136 @@
+// Copyright 2024 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package searchdeployment_test
+
+import (
+	ctx "context"
+	"os"
+	"testing"
+
+	"github.com/aws/aws-sdk-go-v2/service/cloudformation"
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/test/e2e/utility"
+	"github.com/stretchr/testify/assert"
+	"go.mongodb.org/atlas-sdk/v20231115007/admin"
+)
+
+type localTestContext struct {
+	cfnClient      *cloudformation.Client
+	atlasClient    *admin.APIClient
+	projectTmplObj testProject
+	resourceCtx    utility.ResourceContext
+	template       string
+}
+
+type testProject struct {
+	ResourceTypeName string
+	Profile          string
+	ProjectID        string
+	ClusterName      string
+}
+
+const (
+	resourceTypeName  = "MongoDB::Atlas::SearchDeployment"
+	resourceDirectory = "search-deployment"
+	cfnTemplatePath   = "searchdeployment_template.json"
+)
+
+var (
+	profile       = os.Getenv("MONGODB_ATLAS_SECRET_PROFILE")
+	orgID         = os.Getenv("MONGODB_ATLAS_ORG_ID")
+	e2eRandSuffix = utility.GetRandNum().String()
+	stackName     = "stack-searchdeployment-e2e-" + e2eRandSuffix
+	clusterName   = "cfn-e2e-searchdeployment" + e2eRandSuffix
+	projectName   = "cfn-e2e-searchdeployment" + e2eRandSuffix
+)
+
+func TestSearchDeploymentCFN(t *testing.T) {
+	testCtx := setupSuite(t)
+
+	t.Run("Validate Template", func(t *testing.T) {
+		utility.TestIsTemplateValid(t, testCtx.cfnClient, testCtx.template)
+	})
+
+	t.Run("Create Stack", func(t *testing.T) {
+		testCreateStack(t, testCtx)
+	})
+
+	t.Run("Delete Stack", func(t *testing.T) {
```

Comment:
we can include an update check here, mainly just modifying the InstanceSize or NodeCount

Reply:
let's discuss offline, i removed it because i couldn't change NodeCount

Reply:
i will add the update operation, and will change the template so extension is not json